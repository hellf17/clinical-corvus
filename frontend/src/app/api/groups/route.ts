import { auth } from '@clerk/nextjs/server';
import { NextRequest, NextResponse } from 'next/server';

// Simple in-memory rate limiting store (in production, use Redis or similar)
const rateLimitStore = new Map<string, { count: number; timestamp: number }>();

// Rate limiting configuration
const RATE_LIMIT_WINDOW = 60 * 1000; // 1 minute
const RATE_LIMIT_MAX_REQUESTS = 100; // 100 requests per window

// Rate limiting function
function isRateLimited(userId: string): boolean {
  const now = Date.now();
  const key = `${userId}`;
  const record = rateLimitStore.get(key);

  if (!record || now - record.timestamp > RATE_LIMIT_WINDOW) {
    // Reset rate limit counter
    rateLimitStore.set(key, { count: 1, timestamp: now });
    return false;
  }

  if (record.count >= RATE_LIMIT_MAX_REQUESTS) {
    // Rate limit exceeded
    return true;
  }

  // Increment counter
  rateLimitStore.set(key, { count: record.count + 1, timestamp: record.timestamp });
  return false;
}

export async function GET(request: NextRequest) {
  try {
    const { getToken } = await auth();
    const token = await getToken();
    
    if (!token) {
      return NextResponse.json(
        { error: 'Authentication required' },
        { status: 401 }
      );
    }

    // Rate limiting using token as user identifier
    if (isRateLimited(token)) {
      return NextResponse.json(
        { error: 'Rate limit exceeded' },
        { status: 429 }
      );
    }
    
    // Extract query parameters from the request URL
    const { searchParams } = new URL(request.url);
    const queryString = searchParams.toString();
    const url = queryString
      ? `${process.env.NEXT_PUBLIC_BACKEND_URL}/api/groups/?${queryString}`
      : `${process.env.NEXT_PUBLIC_BACKEND_URL}/api/groups/`;
    
    // Forward to backend
    const response = await fetch(url, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${token}`,
      },
    });

    if (!response.ok) {
      return NextResponse.json(
        { error: 'Failed to fetch groups' },
        { status: response.status }
      );
    }

    const data = await response.json();
    return NextResponse.json(data);
  } catch (error) {
    console.error('Error in GET /api/groups:', error);
    return NextResponse.json(
      { error: 'Internal server error' },
      { status: 500 }
    );
  }
}

export async function POST(request: NextRequest) {
  try {
    const { getToken } = await auth();
    const token = await getToken();
    
    if (!token) {
      return NextResponse.json(
        { error: 'Authentication required' },
        { status: 401 }
      );
    }

    // Rate limiting using token as user identifier
    if (isRateLimited(token)) {
      return NextResponse.json(
        { error: 'Rate limit exceeded' },
        { status: 429 }
      );
    }
    
    const body = await request.json();
    
    // Forward to backend
    const response = await fetch(`${process.env.NEXT_PUBLIC_BACKEND_URL}/api/groups/`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${token}`,
      },
      body: JSON.stringify(body),
    });

    if (!response.ok) {
      return NextResponse.json(
        { error: 'Failed to create group' },
        { status: response.status }
      );
    }

    const data = await response.json();
    return NextResponse.json(data);
  } catch (error) {
    console.error('Error in POST /api/groups:', error);
    return NextResponse.json(
      { error: 'Internal server error' },
      { status: 500 }
    );
  }
}