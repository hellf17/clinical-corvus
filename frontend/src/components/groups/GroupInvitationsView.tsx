'use client';

import React, { useState } from 'react';
import { InvitationForm } from '@/components/groups/InvitationForm';
import { InvitationList } from '@/components/groups/InvitationList';
import { Button } from '@/components/ui/Button';
import { Dialog, DialogContent, DialogHeader, DialogTitle, DialogTrigger } from '@/components/ui/Dialog';
import { Plus } from 'lucide-react';

interface GroupInvitationsViewProps {
  groupId: number;
}

export const GroupInvitationsView: React.FC<GroupInvitationsViewProps> = ({ groupId }) => {
  const [showInviteForm, setShowInviteForm] = useState(false);
  const [refreshKey, setRefreshKey] = useState(0);

  const handleInvitationSent = () => {
    setShowInviteForm(false);
    // Trigger a refresh of the invitation list
    setRefreshKey(prev => prev + 1);
  };

  return (
    <div className="space-y-6">
      <div className="flex justify-between items-center">
        <h2 className="text-xl font-bold">Group Invitations</h2>
        <Dialog open={showInviteForm} onOpenChange={setShowInviteForm}>
          <DialogTrigger asChild>
            <Button>
              <Plus className="h-4 w-4 mr-2" />
              Invite Member
            </Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Invite Member to Group</DialogTitle>
            </DialogHeader>
            <InvitationForm 
              groupId={groupId} 
              onSuccess={handleInvitationSent}
              onCancel={() => setShowInviteForm(false)}
            />
          </DialogContent>
        </Dialog>
      </div>

      <InvitationList 
        key={refreshKey}
        groupId={groupId} 
        onInvitationChange={() => setRefreshKey(prev => prev + 1)}
      />
    </div>
  );
};