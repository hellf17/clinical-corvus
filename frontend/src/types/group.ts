/**
 * Types for group management
 */

export type GroupRole = 'admin' | 'member';

export interface Group {
  id: number;
  name: string;
  description?: string;
  max_patients?: number;
  max_members?: number;
  created_at: string;
  updated_at: string;
}

export interface GroupMembership {
  id: number;
  group_id: number;
  user_id: number;
  role: GroupRole;
  joined_at: string;
  invited_by?: number;
}

export interface GroupPatient {
  id: number;
  group_id: number;
  patient_id: number;
  assigned_at: string;
  assigned_by?: number;
}

export interface GroupWithMembersAndPatients extends Group {
  members: GroupMembership[];
  patients: GroupPatient[];
}

export interface GroupCreate {
  name: string;
  description?: string;
  max_patients?: number;
  max_members?: number;
}

export interface GroupUpdate {
  name?: string;
  description?: string;
  max_patients?: number;
  max_members?: number;
}

export interface GroupMembershipCreate {
  user_id: number;
  role: GroupRole;
}

export interface GroupMembershipUpdate {
  role: GroupRole;
}

export interface GroupPatientCreate {
  patient_id: number;
}

export interface GroupWithCounts extends Group {
  member_count?: number;
  patient_count?: number;
}

export interface GroupListResponse {
  items: Group[];
  total: number;
}

export interface GroupWithCountsListResponse {
  items: GroupWithCounts[];
  total: number;
}

export interface GroupMembershipListResponse {
 items: GroupMembership[];
  total: number;
}

export interface GroupPatientListResponse {
  items: GroupPatient[];
  total: number;
}