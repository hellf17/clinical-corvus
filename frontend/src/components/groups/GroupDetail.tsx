'use client';

import React, { useState, useEffect, useCallback } from 'react';
import { GroupWithMembersAndPatients } from '@/types/group';
import { getGroup } from '@/services/groupService';
import { Spinner } from '@/components/ui/Spinner';
import { Alert, AlertDescription } from '@/components/ui/Alert';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/Card';
import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/Tabs';
import { MemberList } from './MemberList';
import { PatientAssignmentList } from './PatientAssignmentList';
import { Button } from '@/components/ui/Button';
import { Edit, Users, UserPlus, FileText } from 'lucide-react';
import Link from 'next/link';
import { useUser } from '@clerk/nextjs';
import { useAuth } from '@clerk/nextjs';
import { canUserManageGroup } from '@/utils/groupPermissions';

interface GroupDetailProps {
  groupId: number;
  onEdit?: () => void;
}

export const GroupDetail: React.FC<GroupDetailProps> = ({ groupId, onEdit }) => {
  const { user } = useUser();
  const { getToken } = useAuth();
  const [group, setGroup] = useState<GroupWithMembersAndPatients | null>(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);

  const fetchGroup = useCallback(async () => {
    try {
      setLoading(true);
      setError(null);
      const groupData = await getGroup(groupId);
      setGroup(groupData);
    } catch (err) {
      setError('Failed to load group details');
      console.error('Error fetching group:', err);
    } finally {
      setLoading(false);
    }
  }, [groupId]);

  useEffect(() => {
    fetchGroup();
  }, [groupId, fetchGroup]);

  if (loading) {
    return (
      <div className="flex justify-center items-center h-64">
        <Spinner className="h-8 w-8" />
      </div>
    );
  }

  if (error) {
    return (
      <Alert variant="destructive">
        <AlertDescription>{error}</AlertDescription>
      </Alert>
    );
  }

  if (!group) {
    return (
      <Alert variant="destructive">
        <AlertDescription>Grupo n√£o encontrado</AlertDescription>
      </Alert>
    );
  }

  return (
    <div className="space-y-6">
      <div className="flex flex-col sm:flex-row justify-between items-start sm:items-center gap-4">
        <div>
          <h1 className="text-2xl font-bold text-gray-900 dark:text-white">{group.name}</h1>
          {group.description && (
            <p className="text-gray-600 dark:text-gray-300 mt-1">{group.description}</p>
          )}
        </div>
        {onEdit && user && canUserManageGroup(group.members.filter(m => m.user_id === Number(user.id)), group.id) && (
          <Button onClick={onEdit} className="flex items-center gap-2">
            <Edit className="h-4 w-4" />
            Editar Grupo
          </Button>
        )}
      </div>

      <Tabs defaultValue="members" className="w-full">
        <TabsList className="grid w-full grid-cols-2">
          <TabsTrigger value="members" className="flex items-center gap-2">
            <Users className="h-4 w-4" />
            Membros ({group.members.length})
          </TabsTrigger>
          <TabsTrigger value="patients" className="flex items-center gap-2">
            <FileText className="h-4 w-4" />
            Pacientes ({group.patients.length})
          </TabsTrigger>
        </TabsList>
        
        <TabsContent value="members" className="mt-6">
          <MemberList
            groupId={group.id}
            members={group.members}
            onMemberAction={fetchGroup}
            currentUserMemberships={user ? group.members.filter(m => m.user_id === Number(user.id)) : []}
          />
        </TabsContent>
        
        <TabsContent value="patients" className="mt-6">
          <PatientAssignmentList 
            groupId={group.id} 
            patients={group.patients} 
            onPatientAction={fetchGroup}
          />
        </TabsContent>
      </Tabs>
    </div>
  );
};