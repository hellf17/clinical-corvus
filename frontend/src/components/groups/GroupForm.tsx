'use client';

import React, { useState } from 'react';
import { GroupCreate, GroupUpdate, Group } from '@/types/group';
import { createGroup, updateGroup } from '@/services/groupService';
import { Button } from '@/components/ui/Button';
import { Input } from '@/components/ui/Input';
import { Textarea } from '@/components/ui/Textarea';
import { Label } from '@/components/ui/Label';
import { Alert, AlertDescription } from '@/components/ui/Alert';
import { useRouter } from 'next/navigation';
import { useAuth } from '@clerk/nextjs';

interface GroupFormProps {
  group?: Group;
  onSuccess?: (group: Group) => void;
  onCancel?: () => void;
}

export const GroupForm: React.FC<GroupFormProps> = ({ group, onSuccess, onCancel }) => {
  const { getToken } = useAuth();
  const [formData, setFormData] = useState<GroupCreate>({
    name: group?.name || '',
    description: group?.description || '',
    max_patients: group?.max_patients || 20, // Default to 20 patients
    max_members: group?.max_members || 10, // Default to 10 doctors
  });
  
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState<string | null>(null);
  const router = useRouter();

  const handleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
    const { name, value } = e.target;
    setFormData(prev => ({
      ...prev,
      [name]: name === 'max_patients' || name === 'max_members' ? parseInt(value) || 0 : value
    }));
  };

  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();
    setLoading(true);
    setError(null);

    
        try {
          
    
          let result: Group;
          
          if (group) {
            // Update existing group
            const updateData: GroupUpdate = {
              name: formData.name,
              description: formData.description,
              // Don't update limits for existing groups
            };
            result = await updateGroup(group.id, updateData);
          } else {
            // Create new group with default limits
            const createData: GroupCreate = {
              name: formData.name,
              description: formData.description,
              max_patients: 20, // Default limit
              max_members: 10,  // Default limit
            };
            result = await createGroup(createData);
          }
          
          if (onSuccess) {
            onSuccess(result);
          } else {
            router.push(`/dashboard-doctor/groups/${result.id}`);
          }
        } catch (err) {
          setError('Failed to save group');
          console.error('Error saving group:', err);
        } finally {
          setLoading(false);
        }
      };
  return (
    <form onSubmit={handleSubmit} className="space-y-6">
      {error && (
        <Alert variant="destructive">
          <AlertDescription>{error}</AlertDescription>
        </Alert>
      )}
      
      <div className="space-y-4">
        <div>
          <Label htmlFor="name">Nome do Grupo *</Label>
          <Input
            id="name"
            name="name"
            value={formData.name}
            onChange={handleChange}
            required
            placeholder="Nome do grupo"
          />
        </div>
        
        <div>
          <Label htmlFor="description">Descrição</Label>
          <Textarea
            id="description"
            name="description"
            value={formData.description || ''}
            onChange={handleChange}
            placeholder="Descrição do grupo"
            rows={3}
          />
        </div>
        
        {/* Hidden fields with default values */}
        <input
          type="hidden"
          name="max_patients"
          value={formData.max_patients}
        />
        <input
          type="hidden"
          name="max_members"
          value={formData.max_members}
        />
      </div>
      
      <div className="flex justify-end gap-3">
        {onCancel && (
          <Button type="button" variant="outline" onClick={onCancel} disabled={loading}>
            Cancelar
          </Button>
        )}
        <Button type="submit" disabled={loading}>
          {loading ? 'Salvando...' : (group ? 'Atualizar Grupo' : 'Criar Grupo')}
        </Button>
      </div>
    </form>
  );
};