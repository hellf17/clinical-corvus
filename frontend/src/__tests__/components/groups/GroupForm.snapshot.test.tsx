import React from 'react';
import { render } from '@testing-library/react';
import { GroupForm } from '@/components/groups/GroupForm';
import { Group } from '@/types/group';

// Mock Next.js router
jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: jest.fn(),
    refresh: jest.fn(),
  }),
}));

describe('GroupForm Snapshot Tests', () => {
  const mockGroup: Group = {
    id: 1,
    name: 'Test Group',
    description: 'Test group description',
    created_at: '2023-01-01T00:00:00Z',
    updated_at: '2023-01-01T00:00:00Z',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot for create form', () => {
    const { asFragment } = render(<GroupForm />);
    
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot for edit form', () => {
    const { asFragment } = render(<GroupForm group={mockGroup} />);
    
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with custom submit text', () => {
    const { asFragment } = render(<GroupForm submitText="Custom Submit" />);
    
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with custom cancel text', () => {
    const { asFragment } = render(<GroupForm cancelText="Custom Cancel" />);
    
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with all custom texts', () => {
    const { asFragment } = render(
      <GroupForm 
        submitText="Save Changes" 
        cancelText="Discard Changes" 
      />
    );
    
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with error state', () => {
    const { asFragment } = render(<GroupForm initialError="This is an error message" />);
    
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with loading state', () => {
    const { asFragment } = render(<GroupForm isLoading={true} />);
    
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with success state', () => {
    const { asFragment } = render(<GroupForm isSuccess={true} successMessage="Operation successful!" />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});