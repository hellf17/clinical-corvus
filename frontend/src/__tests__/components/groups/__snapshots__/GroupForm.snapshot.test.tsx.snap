// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupForm Snapshot Tests matches snapshot for create form 1`] = `
<DocumentFragment>
  <form
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="name"
        >
          Nome do Grupo *
        </label>
        <input
          class="flex h-12 w-full rounded-md border border-input bg-background px-4 py-3 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="name"
          name="name"
          placeholder="Nome do grupo"
          required=""
          value=""
        />
      </div>
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="description"
        >
          Descrição
        </label>
        <textarea
          class="flex min-h-[80px] w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="description"
          name="description"
          placeholder="Descrição do grupo"
          rows="3"
        />
      </div>
      <input
        name="max_patients"
        type="hidden"
        value="20"
      />
      <input
        name="max_members"
        type="hidden"
        value="10"
      />
    </div>
    <div
      class="flex justify-end gap-3"
    >
      <button
        class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
        type="submit"
      >
        Criar Grupo
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`GroupForm Snapshot Tests matches snapshot for edit form 1`] = `
<DocumentFragment>
  <form
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="name"
        >
          Nome do Grupo *
        </label>
        <input
          class="flex h-12 w-full rounded-md border border-input bg-background px-4 py-3 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="name"
          name="name"
          placeholder="Nome do grupo"
          required=""
          value="Test Group"
        />
      </div>
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="description"
        >
          Descrição
        </label>
        <textarea
          class="flex min-h-[80px] w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="description"
          name="description"
          placeholder="Descrição do grupo"
          rows="3"
        >
          Test group description
        </textarea>
      </div>
      <input
        name="max_patients"
        type="hidden"
        value="20"
      />
      <input
        name="max_members"
        type="hidden"
        value="10"
      />
    </div>
    <div
      class="flex justify-end gap-3"
    >
      <button
        class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
        type="submit"
      >
        Atualizar Grupo
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`GroupForm Snapshot Tests matches snapshot with all custom texts 1`] = `
<DocumentFragment>
  <form
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="name"
        >
          Nome do Grupo *
        </label>
        <input
          class="flex h-12 w-full rounded-md border border-input bg-background px-4 py-3 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="name"
          name="name"
          placeholder="Nome do grupo"
          required=""
          value=""
        />
      </div>
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="description"
        >
          Descrição
        </label>
        <textarea
          class="flex min-h-[80px] w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="description"
          name="description"
          placeholder="Descrição do grupo"
          rows="3"
        />
      </div>
      <input
        name="max_patients"
        type="hidden"
        value="20"
      />
      <input
        name="max_members"
        type="hidden"
        value="10"
      />
    </div>
    <div
      class="flex justify-end gap-3"
    >
      <button
        class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
        type="submit"
      >
        Criar Grupo
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`GroupForm Snapshot Tests matches snapshot with custom cancel text 1`] = `
<DocumentFragment>
  <form
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="name"
        >
          Nome do Grupo *
        </label>
        <input
          class="flex h-12 w-full rounded-md border border-input bg-background px-4 py-3 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="name"
          name="name"
          placeholder="Nome do grupo"
          required=""
          value=""
        />
      </div>
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="description"
        >
          Descrição
        </label>
        <textarea
          class="flex min-h-[80px] w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="description"
          name="description"
          placeholder="Descrição do grupo"
          rows="3"
        />
      </div>
      <input
        name="max_patients"
        type="hidden"
        value="20"
      />
      <input
        name="max_members"
        type="hidden"
        value="10"
      />
    </div>
    <div
      class="flex justify-end gap-3"
    >
      <button
        class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
        type="submit"
      >
        Criar Grupo
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`GroupForm Snapshot Tests matches snapshot with custom submit text 1`] = `
<DocumentFragment>
  <form
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="name"
        >
          Nome do Grupo *
        </label>
        <input
          class="flex h-12 w-full rounded-md border border-input bg-background px-4 py-3 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="name"
          name="name"
          placeholder="Nome do grupo"
          required=""
          value=""
        />
      </div>
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="description"
        >
          Descrição
        </label>
        <textarea
          class="flex min-h-[80px] w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="description"
          name="description"
          placeholder="Descrição do grupo"
          rows="3"
        />
      </div>
      <input
        name="max_patients"
        type="hidden"
        value="20"
      />
      <input
        name="max_members"
        type="hidden"
        value="10"
      />
    </div>
    <div
      class="flex justify-end gap-3"
    >
      <button
        class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
        type="submit"
      >
        Criar Grupo
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`GroupForm Snapshot Tests matches snapshot with error state 1`] = `
<DocumentFragment>
  <form
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="name"
        >
          Nome do Grupo *
        </label>
        <input
          class="flex h-12 w-full rounded-md border border-input bg-background px-4 py-3 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="name"
          name="name"
          placeholder="Nome do grupo"
          required=""
          value=""
        />
      </div>
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="description"
        >
          Descrição
        </label>
        <textarea
          class="flex min-h-[80px] w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="description"
          name="description"
          placeholder="Descrição do grupo"
          rows="3"
        />
      </div>
      <input
        name="max_patients"
        type="hidden"
        value="20"
      />
      <input
        name="max_members"
        type="hidden"
        value="10"
      />
    </div>
    <div
      class="flex justify-end gap-3"
    >
      <button
        class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
        type="submit"
      >
        Criar Grupo
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`GroupForm Snapshot Tests matches snapshot with loading state 1`] = `
<DocumentFragment>
  <form
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="name"
        >
          Nome do Grupo *
        </label>
        <input
          class="flex h-12 w-full rounded-md border border-input bg-background px-4 py-3 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="name"
          name="name"
          placeholder="Nome do grupo"
          required=""
          value=""
        />
      </div>
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="description"
        >
          Descrição
        </label>
        <textarea
          class="flex min-h-[80px] w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="description"
          name="description"
          placeholder="Descrição do grupo"
          rows="3"
        />
      </div>
      <input
        name="max_patients"
        type="hidden"
        value="20"
      />
      <input
        name="max_members"
        type="hidden"
        value="10"
      />
    </div>
    <div
      class="flex justify-end gap-3"
    >
      <button
        class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
        type="submit"
      >
        Criar Grupo
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`GroupForm Snapshot Tests matches snapshot with success state 1`] = `
<DocumentFragment>
  <form
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="name"
        >
          Nome do Grupo *
        </label>
        <input
          class="flex h-12 w-full rounded-md border border-input bg-background px-4 py-3 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="name"
          name="name"
          placeholder="Nome do grupo"
          required=""
          value=""
        />
      </div>
      <div>
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="description"
        >
          Descrição
        </label>
        <textarea
          class="flex min-h-[80px] w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="description"
          name="description"
          placeholder="Descrição do grupo"
          rows="3"
        />
      </div>
      <input
        name="max_patients"
        type="hidden"
        value="20"
      />
      <input
        name="max_members"
        type="hidden"
        value="10"
      />
    </div>
    <div
      class="flex justify-end gap-3"
    >
      <button
        class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
        type="submit"
      >
        Criar Grupo
      </button>
    </div>
  </form>
</DocumentFragment>
`;
